### ISSUE TITLE
Memory leak

### ISSUE THREAD

**Issue by khasanakv** [amplitude/Amplitude-Swift#230]
## Current Behavior
Memory increase while sending analytics

![image](https://github.com/user-attachments/assets/06fa26a0-1b03-405c-bfdf-1571b99953b7)

send analytics parallel memory just increasing 

## Environment
- SDK Version:1.9.2
- OS Info: iOS 17.5

**Comment by crleona** [comment]
Hi @khasanakv are you able to provide more details and confirm that you're using 1.9.2? We fixed a similar issue in https://github.com/amplitude/Amplitude-Swift/pull/221, which should be included in 1.9.2.

**Comment by YunusDeveloper** [comment]
I also reproduce this memory leak in version 1.9.2. 
 ```
let amplitudeConfig = Configuration(
            apiKey: myApiKey,
            flushIntervalMillis: 1,
            autocapture: [.appLifecycles, .elementInteractions, .elementInteractions, .sessions]
 )
```
when I configure Amplitude and adding options if add .sessions option memory leak reproduces if I remove no it doesn't happen

**Comment by crleona** [comment]
Well, we don't appear to have a leak, but we are hanging on to memory longer than we need it - this should be fixed in https://github.com/amplitude/Amplitude-Swift/pull/232. Thanks @khasanakv for the report and @YunusDeveloper for the repro steps.

**Comment by crleona** [comment]
The fix has been released in 1.9.3
